web/src/api/images/get_unverified_images.ts

```diff
@@ -0,0 +1,176 @@
+import {
+  errorHasuraQuery,
+  errorNotAllowed,
+  errorResponse,
+} from "src/backend/errors";
+import { NextApiRequest, NextApiResponse } from "next";
+import * as yup from "yup";
+import { protectInternalEndpoint } from "src/backend/utils";
+import { getAPIServiceGraphqlClient } from "src/backend/graphql";
+import { getSdk as getUnverifiedImagesSDK } from "@/api/images/graphql/getUnverifiedImages.generated";
+import { S3Client, GetObjectCommand } from "@aws-sdk/client-s3";
+import { getSignedUrl } from "@aws-sdk/s3-request-presigner";
+import { logger } from "@/lib/logger";
+
+export type ImageGetAllUnverifiedImagesResponse = {
+  logo_img_url?: string;
+  hero_image_url?: string;
+  showcase_img_urls?: string[];
+};
+
+/**
+ * Used when an app is loaded to show all unverified images
+ * @param req
+ * @param res
+ */
+export const handleGetAllUnverifiedImages = async (
+  req: NextApiRequest,
+  res: NextApiResponse
+) => {
+  try {
+    if (!protectInternalEndpoint(req, res)) {
+      return;
+    }
+
+    if (req.method !== "GET") {
+      return errorNotAllowed(req.method, res, req);
+    }
+    const body = JSON.parse(req.body);
+    if (body?.action.name !== "get_all_unverified_images") {
+      return errorHasuraQuery({
+        res,
+        req,
+        detail: "Invalid action.",
+        code: "invalid_action",
+      });
+    }
+
+    if (body.session_variables["x-hasura-role"] === "admin") {
+      return errorHasuraQuery({
+        res,
+        req,
+        detail: "Admin is not allowed to run this query.",
+        code: "admin_not_allowed",
+      });
+    }
+    const userId = body.session_variables["x-hasura-user-id"];
+    if (!userId) {
+      return errorHasuraQuery({
+        res,
+        req,
+        detail: "userId must be set.",
+        code: "required",
+      });
+    }
+
+    const teamId = body.session_variables["x-hasura-team-id"];
+    if (!teamId) {
+      return errorHasuraQuery({
+        res,
+        req,
+        detail: "teamId must be set.",
+        code: "required",
+      });
+    }
+
+    const app_id = req.query.app_id;
+    if (!app_id) {
+      return errorHasuraQuery({
+        res,
+        req,
+        detail: "app_id must be set.",
+        code: "required",
+      });
+    }
+    const client = await getAPIServiceGraphqlClient();
+    const { app: appInfo } = await getUnverifiedImagesSDK(
+      client
+    ).GetUnverifiedImages({
+      team_id: teamId,
+      app_id: app_id as string,
+      user_id: userId,
+    });
+    if (appInfo.length === 0 || appInfo[0].app_metadata.length === 0) {
+      return errorHasuraQuery({
+        res,
+        req,
+        detail: "User does not have access to this app.",
```

Comment:
In general is best to return more generic errors for permissions, e.g. "App not found"